pkg/provider/azure/keyvault/keyvault.go

```diff
@@ -151,10 +153,41 @@ func (a *Azure) ValidateStore(store esv1beta1.GenericStore) error {
 	return nil
 }
 
-// Empty GetAllSecrets.
+// Implements store.Client.GetAllSecrets Interface.
+// Retrieves a map[string][]byte with the secret names as key and the secret itself as the calue.
 func (a *Azure) GetAllSecrets(ctx context.Context, ref esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
-	// TO be implemented
-	return nil, fmt.Errorf("GetAllSecrets not implemented")
+	basicClient := a.baseClient
+	secretsMap := make(map[string][]byte)
+	checkTags := len(ref.Tags) > 0
+	checkName := ref.Name != nil && len(ref.Name.RegExp) > 0
+
+	secretListIter, err := basicClient.GetSecretsComplete(context.Background(), *a.provider.VaultURL, nil)
```

Comment:
In the `ExternalSecretFind` there is a new field called `path` used to reduce the amount of secrets handled by this call (specially to avoid external-secrets getting more information than the user might want). I believe we can use it here after applying some logic if it's not nil.

Reply:
Nevermind, just looked at GetSecretsComplete implementation, it seems that this VaultURL can only be the root path indeed. Maybe we should just add an error if someone used `find.path` with azure Keyvault provider.